hubploy/auth.py

```diff
@@ -54,7 +54,11 @@ def registry_auth_aws(deployment, project, service_key):
 
     This changes *global machine state* on where docker can push to!
     """
-    # This assumes that AWS_ACCESS_KEY_ID and AWS_SECRET_KEY_ID are in env vars
+    service_key_path = os.path.join(
+        'deployments', deployment, 'secrets', service_key
+    )
+    os.environ['AWS_SHARED_CREDENTIALS_FILE'] = os.path.abspath(service_key_path)
```

Comment:
is there a way to specify this explicitly in the call to the boto client constructor? I'd prefer that if possible.

Reply:
I think it would work to parse the config file and pass the keys in as arguments. Something like:

```python
from ConfigParser import ConfigParser
config = ConfigParser()
config.read(service_key_path)

client = boto3.client('ecr', **dict(config['default'])
```